src/ifaddrs.rs

```diff
@@ -42,11 +42,43 @@ cfg_if! {
     }
 }
 
+/// Workaround a bug in XNU where netmasks will always have the wrong size in
+/// the sa_len field due to the kernel ignoring trailing zeroes in the structure
+/// when setting the field. See https://github.com/nix-rust/nix/issues/1709#issuecomment-1199304470
+///
+/// To fix this, we overwrite `sa_len` to have the known (static) value of the
+/// structure.
+#[cfg(target_os = "macos")]
+fn workaround_xnu_bug(info: &libc::ifaddrs) {
+    let sock = info.ifa_netmask;
+    if sock.is_null() {
+        return
+    }
+    let family = unsafe {
+        (*sock).sa_family
+    };
+
+    let fixup_size = match i32::from(family) {
+        libc::AF_INET => std::mem::size_of::<libc::sockaddr_in>(),
+        libc::AF_INET6 => std::mem::size_of::<libc::sockaddr_in6>(),
+        _ => return
+    };
+
+    assert!(fixup_size < u8::MAX.into());
+    unsafe {
+        (*sock).sa_len = (*sock).sa_len.max(fixup_size as u8);
+    }
```

Comment:
You can avoid an extra explicit assertion like this:
```suggestion
    unsafe {
        (*sock).sa_len = (*sock).sa_len.max(u8::try_from(fixup_size).unwrap());
    }
```
However, this isn't safe.  How do you know that the memory past sa_len is actually allocated and unused?  You can't.  Instead, you should fix it by copying it to a new location.